### ISSUE TITLE
NumberPerfectGroups(1) return 0, not 1

### ISSUE THREAD

**Issue by fingolfin** [gap-system/gap#1052]
The documentation says for `IsPerfectGroup`: "A group is perfect if it equals its derived subgroup" and that is also the definition I know. Hence, the trivial group is perfect.

However, `NumberPerfectGroups(1)` return 0, not 1. I see two solutions:

1. Change `NumberPerfectGroups` to return 1 (for people who want to use the perfect group library, there is still `NumberPerfectLibraryGroups`; also, it would be trivial to extend the perfect group library to contain the trivial group, if we really wanted to).

2. Change the documentation of `NumberPerfectGroups` to explicitly state that the trivial group is excluded.

Personally, I prefer 1 simply because the function is named and`NumberPerfectGroups` and not `NumberNontrivialPerfectGroups`. Alas, for reasons of backwards compatibility it may be the only feasible route.

**Comment by fingolfin** [comment]
Assigning to @hulpke as I think he wrote the relevant code (?) and in general probably knows most about this, i.e. what solution is best / most realistic (perhaps one I didn't even think of :-)

**Comment by ChrisJefferson** [comment]
I agree. It would be nice if `NumberPerfectGroups` lined up with the behaviour of `IsPerfectGroup`, and (of course) `IsPerfectGroup(Group(()))` returns `true`.

**Comment by hulpke** [comment]
The code was written by Volkmar Flesh who felt very strongly about implementing exactly the listing from the Plesken/Holt book, that is excluding the trivial group, even if it does not make sense. (This is also the reason for the fp representation being the default, as the groups are given in the book by presentations.) I've removed myself as assignee, as I've never really looked into this code, but I certainly support including the trivial group.

**Comment by fingolfin** [comment]
@hulpke OK, thanks for the background. My assumption was based on these commits from the old repos:
```
* commit 74e210043edda6c349ae7099e78962434e209505
| Author: Alexander Hulpke <hulpke@math.colostate.edu>
| Date:   2004-07-22 17:31:36 +0000
|
|     Eliminated the trivial group from perfect groups library. AH
|
* commit 9cd3a1c0771136e35b59780b4989a1921d9d4807
| Author: Alexander Hulpke <hulpke@math.colostate.edu>
| Date:   2000-01-13 19:20:22 +0000
|
|     Added trivial group for completeness. AH
|
* commit de69f6f2bbced4211c1c79f60532bec3856d115a
| Author: Alexander Hulpke <hulpke@math.colostate.edu>
| Date:   1999-08-16 09:56:08 +0000
|
|     PerfectGroup(1,1) works.
|
* commit ee8483303f3c48dcc20412a4f6c1e7dc37135bbf
  Author: Alexander Hulpke <hulpke@math.colostate.edu>
  Date:   1997-01-20 15:46:12 +0000

      Added a first version of the perfect groups library
```
But I guess what I see here is  you adding the trivial group, and then somebody (Volkmar Felsch?) complaining, after which you removed it again :-).

**Comment by hulpke** [comment]
@fingolfin 
Thanks -- I clearly do not remember the details (and 2004 is before my searchable email history), so I do not recall the reason for the 2004 reversion.

I've things in to allow the trivial perfect group again. (https://github.com/gap-system/gap/pull/1046/commits/26ba7c2a126307fca5b5bdb51d1fc55e67f6bb53
as part of #1046)

**Comment by alex-konovalov** [comment]
This is resolved by #1046 (commit https://github.com/gap-system/gap/pull/1046/commits/ed74194586963422ec42d197a8f5becdfb670e47)
